app/models/piece.rb

```diff
@@ -1,2 +1,40 @@
 class Piece < ApplicationRecord
+  def is_obstructed?(new_x, new_y)  
+    horizontal_obstructed?(new_x)
+    vertical_obstructed?(new_y)
+    diagonal_obstructed?(new_x, new_y)
+    invalid_move?(new_x, new_y)
+  end 
+  
+  def horizontal_obstructed?(new_x) #Horizontal movement only
+    Piece.where(x_position:(self.x_position + 1..new_x),
+                y_position:self.y_position).present?
+  end
+  
+  def vertical_obstructed?(new_y)  #Vertical movement only
+    Piece.where(y_position:(self.y_position + 1..new_y), 
+                x_position:self.x_position).present?
+  end
+  
+
+  def diagonal_obstructed?(new_x, new_y)  #Diagonal movement only
+    Piece.where(y_position:(self.y_position + 1..new_y),
```

Comment:
This one is a little tricky, I don't think this is going to make the query you want. That's okay though lets get this one merged and we can make a bugfix next.